croncat/src/system/mod.rs

```diff
@@ -238,32 +267,58 @@ pub async fn run_retry(
     chain_id: &String,
     shutdown_tx: &ShutdownTx,
     config: &ChainConfig,
-    factory: Arc<Mutex<Factory>>,
-    agent: Arc<Agent>,
-    manager: Arc<Manager>,
-    tasks: Arc<Mutex<Tasks>>,
+    factory: &Arc<Mutex<Factory>>,
+    agent: &Arc<Agent>,
+    manager: &Arc<Manager>,
+    tasks: &Arc<Mutex<Tasks>>,
 ) -> Result<(), Report> {
-    // TODO: Rethink this retry logic
-    // let retry_strategy = FixedInterval::from_millis(5000).take(1200);
-
-    // Retry::spawn(retry_strategy, || async {
-    run(
-        chain_id,
-        shutdown_tx,
-        config,
-        factory,
-        agent,
-        manager,
-        tasks,
+    // TODO: What's the strategy for retrying?
+    let retry_strategy = FixedInterval::from_millis(5000).take(1200);
```

Comment:
could be nice exponential backoff someday

Reply:
Yes!